pkg-manager/package-requester/src/packageRequester.ts

```diff
@@ -341,11 +341,33 @@ function fetchToStore (
     opts.fetchRawManifest = true
   }
 
+  let promiseSharer = pShare
+
   if (!ctx.fetchingLocker.has(opts.pkg.id)) {
     const fetching = pDefer<PkgRequestFetchResult>()
     const { filesIndexFile, target } = getFilesIndexFilePath(ctx, opts)
 
-    doFetchToStore(filesIndexFile, fetching, target) // eslint-disable-line
+    // Most fetchers (e.g. tarball downloader) are kicked off immediately
+    // so the data is fully ready once requested.
+    // However, directory fetcher should defer loading until actually requested,
+    // in case local files are necessary (e.g. symlinked local packages)
+    let blocker: Promise<void>
+
+    if (opts.pkg.resolution.type === 'directory') {
+      const blocked: pDefer.DeferredPromise<void> = pDefer()
```

Comment:
I'm not in love with this implementation, but basically we want to block the execution of `doFetchToStore` until actually invoked

_only_ apply this to directory fetchers though, others like tarball should still get kicked off.

Alternative approach is instead of passing the `blocker` promise, invoke `doFetchToStore(filesIndexFile, fetching, target, blocker)` in the `promiseSharer` callback... Would need to be careful about deduping